src/Core/Framework/Api/ApiException.php

```diff
@@ -322,10 +322,23 @@ public static function invalidSchemaDefinitions(string $filename, \JsonException
         );
     }
 
+    /**
+     * @deprecated tag:v6.7.0 - Will return status code 400 instead of 500
+     */
     public static function invalidAccessKey(): self
     {
+        if (!Feature::isActive('v6.7.0.0')) {
+            Feature::triggerDeprecationOrThrow('v6.7.0.0', 'Will return status code 400 instead of 500');
```

Comment:
We should not trigger deprecations when we call the code internally, because externals cannot fix that. To stop phpstan complaining you can use one of the reasons from: https://github.com/shopware/shopware/blob/c4deb84620cdb8f815a0a8ececc0b09da88dc0a8/src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Deprecation/DeprecatedMethodsThrowDeprecationRule.php#L56 - probably `reason:exception-change` makes sense here.